core/slice.go

```diff
@@ -294,8 +294,8 @@ func (sl *Slice) CollectSubRollups(b *types.Block) ([]types.Transactions, error)
 			// Look for pending ETXs first in pending ETX cache, then in database
 			if res, ok := sl.pendingEtxs.Get(hash); ok && res != nil {
 				pendingEtxs = res.([]types.Transactions)
-			} else if res := rawdb.ReadPendingEtxs(sl.sliceDb, hash); res != nil {
-				pendingEtxs = res
+			} else if res := rawdb.ReadPendingEtxs(sl.sliceDb, hash); res.Header != nil {
```

Comment:
This is a weird check.. I would prefer if `ReadPendingEtxs()` returned a pointer and nil checked against that. I suppose this works though? Just weird..

Reply:
Changed it